google/cloud/aiplatform/training_jobs.py

```diff
@@ -1201,6 +1232,43 @@ def __init__(
                 "set using aiplatform.init(staging_bucket='gs://my-bucket')"
             )
 
+    @classmethod
+    def get(
+        cls,
+        resource_name: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> "CustomTrainingJob":
+        """Get CustomTrainingJob for the given resource_name.
+
+        Raises:
+            ValueError: If the retrieved training job's training task definition
+                doesn't match the custom training task definition.
+
+        Returns:
+            An AI Platform Training Job
+        """
+
+        # Create job with dummy parameters
+        # These parameters won't be used as user can not run the job again.
+        # If they try, an exception will be raised.
+        self = cls._empty_constructor()
```

Comment:
`project`, `location`, and `credentials` are retrieved from the global config but this can cause a failure if there's a mismatch with the passed in `resource name`

It looks like `resource_name` can be a resource name or a resource id.

If it's a resource name then the project and location should be extracted and validated against the project and location passed in if not None.

If it's a resource id then pass in project, location, credentials.

Reply:
Got it.

Reply:
@sasha-gitg I see the potential for conflict with the passed in project vs the project in the `resource_name`. 

However, isn't that  issue in `Dataset.__init__` and `Model.__init__` as well?

Reply:
That's a good point. For now, we can keep this consistent with Dataset/Model/Endpoint and pass in project, location, credentials as passed in to the method. Can you please open a ticket to extract and initialize the client with the extracted project and location when receiving a fully qualified resource name in all cases in the SDK?